packages/kbn-apm-config-loader/src/config.ts

```diff
@@ -0,0 +1,105 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { join } from 'path';
+import { merge, get } from 'lodash';
+import { execSync } from 'child_process';
+// deep import to avoid loading the whole package
+import { getDataPath } from '@kbn/utils/target/path';
+import { readFileSync } from 'fs';
+
+const defaultConfig = {
+  active: false,
+  serverUrl: 'https://f1542b814f674090afd914960583265f.apm.us-central1.gcp.cloud.es.io:443',
+  // The secretToken below is intended to be hardcoded in this file even though
+  // it makes it public. This is not a security/privacy issue. Normally we'd
+  // instead disable the need for a secretToken in the APM Server config where
+  // the data is transmitted to, but due to how it's being hosted, it's easier,
+  // for now, to simply leave it in.
+  secretToken: 'R0Gjg46pE9K9wGestd',
+  globalLabels: {},
+  breakdownMetrics: true,
+  centralConfig: false,
+  logUncaughtExceptions: true,
+};
+
+export class ApmConfiguration {
+  constructor(
+    private readonly rootDir: string,
+    private readonly rawKibanaConfig: Record<string, any>
+  ) {}
+
+  public getConfig(serviceName: string) {
+    const apmConfig = merge(defaultConfig, this.getDevConfig());
+
+    const rev = this.getGitRev();
+    if (rev !== null) {
+      apmConfig.globalLabels.git_rev = rev;
+    }
+
+    const uuid = this.getKibanaUuid();
+    if (uuid) {
+      apmConfig.globalLabels.kibana_uuid = uuid;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { version } = require(join(this.rootDir, 'package.json'));
+
+    return {
+      ...apmConfig,
+      serviceName: `${serviceName}-${version.replace(/\./g, '_')}`,
+    };
+  }
+
+  private getKibanaUuid() {
+    // try to access the `server.uuid` value from the config file first.
+    // if not manually defined, we will then read the value from the `{DATA_FOLDER}/uuid` file.
+    // note that as the file is created by the platform AFTER apm init, the file
+    // will not be present at first startup, but there is nothing we can really do about that.
+    if (get(this.rawKibanaConfig, 'server.uuid')) {
+      return this.rawKibanaConfig.server.uuid;
+    }
+
+    const dataPath: string = get(this.rawKibanaConfig, 'path.data') || getDataPath();
+    try {
+      const filename = join(dataPath, 'uuid');
+      return readFileSync(filename, 'utf-8');
+    } catch (e) {} // eslint-disable-line no-empty
+  }
+
+  private getDevConfig() {
+    try {
+      const apmDevConfigPath = join(this.rootDir, 'config', 'apm.dev.js');
+      return require(apmDevConfigPath);
+    } catch (e) {
+      return {};
+    }
+  }
```

Comment:
I'm guessing this function would be replaced when we add config keys for APM to the regular yaml file to read from `this.rawKibanaConfig` as well?

Reply:
Not sure how exactly the apm team was using this, but if it's really just for local development, I guess that once the properties are exposed / consumed from the `kibana.yaml` config file, the dev config can just go away and be replaced by local `kibana.yaml` config overrides? Need confirmation from apm though.